sklearn/metrics/scorer.py

```diff
@@ -126,7 +126,13 @@ def __call__(self, clf, X, y, sample_weight=None):
         y_type = type_of_target(y)
         y_pred = clf.predict_proba(X)
         if y_type == "binary":
-            y_pred = y_pred[:, 1]
+            if y_pred.shape[1] == 2:
+                y_pred = y_pred[:, 1]
+            else:
+                raise ValueError('got predict_proba of shape;'
```

Comment:
you're not actually providing the shape.

Reply:
ok, what is happening is this:
1.  previously, when there was only one classifier, but expecting two, the y_pred was not resolving.   This was the error:  `IndexError: index 1 is out of bounds for axis 1 with size 1`
2.  so, we made a check and an adjustment so that if the shape was 2 (meaning it only returned one column vector probabilities, which are all 1's), we want to raise a `ValueError`.  
3.  What should we call it? I don't think returning the shape (even if we did that) is informative.  
How about:
- "vector of probabilities returned`predict_proba` is all 1's because only one classifier is in the model."

Does that work?